dist/$rsi.schema.json

```diff
@@ -0,0 +1,224 @@
+        },
+        "resources": {
+          "$ref": "#/definitions/resourcesDefinition"
+        },
+        "types": {
+          "$ref": "#/definitions/typesDefinition"
+        },
+        "description": {
+          "type": "string"
+        },
+        "$schema": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "description",
+        "name",
+        "$schema"
+      ],
+      "additionalProperties": false
+    },
+    "resourcesDefinition": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "objectModel": {
+          "$ref": "#/definitions/objectModelDefinition"
+        }
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": true
+    },
+    "typesDefinition": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": true
+    },
+    "objectModelDefinition": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "object",
+          "properties": {
+            "description": {
+              "type": "string",
+              "minLength": 2
+            },
+            "type": {
+              "type": "string",
+              "enum": [
+                "string",
+                "number",
+                "integer",
+                "boolean",
+                "array",
+                "object"
+              ]
+            },
+            "format": {
+              "type": "string",
+              "pattern": "uuid"
+            }
+          },
+          "required": [
+            "description",
+            "type",
+            "format"
+          ]
+        },
+        "name": {
+          "type": "string"
+        },
+        "uri": {
+          "type": "object",
+          "properties": {
+            "description": {
+              "type": "string",
+              "minLength": 2
+            },
+            "type": {
+              "type": "string",
+              "pattern": "string"
+            },
+            "format": {
+              "type": "string",
+              "pattern": "uri"
+            }
+          },
+          "required": [
+            "description",
+            "type",
+            "format"
+          ]
+        }
+      },
+      "patternProperties": {
+        "^[a-zA-Z0-9]+$": {
+          "$ref": "#/definitions/objectPropertyDefinition"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "id",
+        "name",
+        "uri"
+      ]
+    },
+    "objectModelReference": {
+      "type": "object",
+      "properties": {
+        "#ref": {
+          "type": "string",
+          "pattern": "^([a-zA-Z]+\\.[a-zA-Z]+\\.[a-zA-Z]+Object|[a-zA-Z]+\\.[a-zA-Z]+Type)$"
```

Comment:
the regex definition looks so different compared to the strange `patternProperties`.

https://github.com/wzr1337/rsi.schema/pull/1/files#diff-d04eb47088d1746cdef153bc7e71c9afR123